### PR TITLE
WIP [for 0.9.0] New file for `project settings` and its location in TOC

### PR DESCRIPTION
**Pull request by j-c-berger**
Signed-off-by: Jacob Berger <jacob.berger@ibm.com>

PR for Issue # 1850: https://github.com/eclipse/codewind/issues/1850

@rajivnathan and/ or @ssh24 please review the new file I created, `Project settings` that combined the docs from 1850. 

@micgibso, please review the TOC changes. 

Thanks--JB

### CODE REVIEW

File: docs/_documentations/projects-settings.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+* [Maven profiles](#maven-profiles)
+* [Maven properties](#maven-properties)
+* [Paths to ignore for file changes](#paths-to-ignore-for-file-changes)
+* [Project status ping timeout](#project-status-ping-timeout)
+
+## **Context root**
+`contextRoot: <string>`
+- Codewind uses the value to determine the project state.
+- When the Open Application action is used, it uses the value as the initial endpoint. 
+- If an incorrect context root is set, the project is stuck in the starting state.
```
Comment (ssh24) [inline docs/_documentations/projects-settings.md]: Can we explain here why the project is stuck at starting state? The reason is: If an incorrect context root is set, the project will be stuck pinging the wrong the endpoint and the project won't come up as started

File: docs/_documentations/projects-settings.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+- If the health check endpoint is set, the context root is not used to determine the project state.
+- If the value is not set, the default value is `/`.
+
+## **Health check endpoint**
+`healthCheck: <string>`
+- Codewind uses the value to determine the project state.
+- Codewind expects to use this value for the application health check endpoint.
+- If a wrong health check endpoint is set, the project is stuck in the starting state.
```
Comment (ssh24) [inline docs/_documentations/projects-settings.md]: Same reason here as the context root

File: docs/_documentations/projects-settings.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+- If the health check endpoint is set, the context root is not used to determine the project state.
```
Comment (ssh24) [inline docs/_documentations/projects-settings.md]: Can we also mention the default value here? I think the default value is `/` (correct me if I am wrong)

File: docs/_documentations/projects-settings.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+## HTTPS application
+`isHttps: <boolean>`
+- This value tells Codewind to use the HTTPS protocol when Codewind detects the application status, and also when Codewind launches the application in a browser.
+- If your application supports HTTPS, set `isHttps` to `true`, and Codewind uses HTTPS instead of HTTP to detect the application status and to open browser URLs.
+- The default value of this setting is `false`.
+
+## **Internal application port**
+`internalPort: <string>`
+- When exposed, Codewind does not automatically expose the port.
```
Comment (ssh24) [inline docs/_documentations/projects-settings.md]: Could we reword this? I don't know what you meant by `Codewind does not automatically expose the port`

File: docs/_documentations/projects-settings.md [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+`isHttps: <boolean>`
+- This value tells Codewind to use the HTTPS protocol when Codewind detects the application status, and also when Codewind launches the application in a browser.
+- If your application supports HTTPS, set `isHttps` to `true`, and Codewind uses HTTPS instead of HTTP to detect the application status and to open browser URLs.
+- The default value of this setting is `false`.
+
+## **Internal application port**
+`internalPort: <string>`
+- When exposed, Codewind does not automatically expose the port. 
+- Codewind uses the value in conjunction with the context root to determine the project state.
+- If an incorrect port is set, the project is stuck in starting state.
```
Comment (ssh24) [inline docs/_documentations/projects-settings.md]: Similar description here as the context root, health check

File: docs/_documentations/projects-settings.md (hunk 4 above)
Comment (j-c-berger) [reply docs/_documentations/projects-settings.md]: I took this sentence to mean that Codewind does not provide (or "expose") the port, but rather, users would need to find the port themselves.

**Comment by j-c-berger** [comment]
@ssh24, thanks for the feedback! I made most of the changes (see my response to your one comment above) and made a new commit, "revised content for project settings." It's ready for you whenever you get a chance. thanks!

File: docs/_documentations/projects-settings.md (hunk 4 above)
Comment (ssh24) [reply docs/_documentations/projects-settings.md]: For internal port codewind (turbine) gets the port dynamically from the container information. So by default the port is the same as the one exposed in Dockerfile or the default container port. The user can specify their own port but they would need to also make sure it is exposed on the Dockerfile.

Comment (j-c-berger) [reply docs/_documentations/projects-settings.md]: Thanks, Sakib, I included what you put here in the new commit. See 'revised internal application port.`

**Pull request merged**